### PR TITLE
Merge SpatialGDKEditorToolbarSettings into USpatialGDKEditorSettings

### PR DESCRIPTION
**Pull request by mattyoung-improbable**
#### Description
Settings Sections can only map to a single UObject. Currently the settings in USpatialGDKEditorSettings were never being set via the settings window because the child class SpatialGDKEditorToolbarSettings was overriding it.

In this PR: Merged all the settings into the EditorSettings, reorganise them into more logical categories, and set their default values.
Fixes: UNR-860

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 1]
```diff
@@ -4,21 +4,30 @@
...
 
 	return FString::Format(TEXT(
-		"SpatialOSLaunchArgument={0}, "
-		"SpatialOSSnapshotPath={1}, "
-		"GeneratedSchemaOutputFolder={2}, ")
+		"ProjectRootFolder={0}, "
+		"SpatialOSLaunchArgument={1}, "
+		"bStopSpatialOnExit={2}, "
+		"SpatialOSSnapshotPath={3}, "
+		"SpatialOSSnapshotFile={4}")
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
		"SpatialOSSnapshotFile={4}, "
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 2]
```diff
@@ -4,21 +4,30 @@
...
+		"GeneratedSchemaOutputFolder={5}, "
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
		"GeneratedSchemaOutputFolder={5}")
```

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 3]
```diff
@@ -37,11 +50,15 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
...
-	FORCEINLINE FString GetGeneratedSchemaOutputFolder() const
+	FORCEINLINE FString GetSpatialOSLaunchConfig() const
 	{
-		return GeneratedSchemaOutputFolder.Path.IsEmpty()
-			? FPaths::ConvertRelativePathToFull(FPaths::Combine(GetSpatialOSDirectory(), FString(TEXT("schema/unreal/generated/"))))
-			: GeneratedSchemaOutputFolder.Path;
+		return SpatialOSLaunchConfig.IsEmpty() ? FString(TEXT("default_launch.json")) : SpatialOSLaunchConfig;
+	}
+
+	UFUNCTION()
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]: Why was this made a UFUNCTION()?

Comment (mattyoung-improbable) [reply SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]: Original definition in SpatialGDKEditorToolbarSettings.h (deleted in this PR) had defined it as a UFUNCTION, probably save to delete.

**Pull request merged**